### PR TITLE
Enable full suite of integration tests to be run in parallel via pytest

### PR DESCRIPTION
**Pull request by biosafetylvl5**
Relevant XKCD:

tests!!!! :)) 

* [x] Required ***existing tests*** pass (ie full_test.sh, others as appropriate)
* [x] NO REQUIRED ***unit tests*** (explain why not required)
* [x] NO REQUIRED ***integration tests***
* [x] NO REQUIRED ***documentation*** (explain why not required)
* [x] Required ***release notes*** added for new/modified functionality
* [x] NO REQUIRED ***updates to other repos*** (explain why not required)

# Testing Instructions
- Install geoips (or just update installed dependencies)
- Run without paralellization: `pytest -m integration` (tests marked as `integration` and set to *not* run by default)
- Run `pytest -n {number of core you want to use, "auto" for all} -m integration`
- 🙂 
- Warning: tests will fail because of memory limitations if you try to run more runners than your machine can handle.

# Summary
- Runs the scripts called by `tests/integration_tests/full_test.sh` in pytest and uses [pytest-xdist](https://pypi.org/project/pytest-xdist/) to parallelize
- ~5X improvement on winter (a CIRA server) ~45m => 8m using 6 workers

# Example Output

``` bash
geoips_user@8f19a9c9d3c4:/app/geoips_packages/geoips/tests/integration_tests$ pytest -n auto -m integration
================================================================ test session starts ================================================================
platform linux -- Python 3.11.10, pytest-8.2.2, pluggy-1.5.0 -- /usr/local/bin/python3
cachedir: .pytest_cache
rootdir: /app/geoips_packages/geoips
configfile: pytest.ini
plugins: cov-5.0.0, xdist-3.6.1
6 workers [53 items]    
scheduling tests via LoadScheduling

test_full_test.py::test_integ_test_script[$geoips_repopath/tests/scripts/console_script_create_sector_image.sh] 
test_full_test.py::test_integ_test_script[$geoips_repopath/docs/build_docs.sh $data_fusion_repopath $data_fusion_pkgname html_only] 
test_full_test.py::test_integ_test_script[$geoips_repopath/tests/scripts/amsr2.config_based_overlay_output.sh] 
test_full_test.py::test_integ_test_script[$geoips_repopath/docs/build_docs.sh $geoips_repopath $geoips_pkgname html_only] 
test_full_test.py::test_integ_test_script[$geoips_repopath/tests/scripts/console_script_list_available_plugins.sh] 
test_full_test.py::test_integ_test_script[$geoips_repopath/tests/utils/check_code.sh all $recenter_tc_repopath] 
[gw1] [  1%] FAILED test_full_test.py::test_integ_test_script[$geoips_repopath/tests/scripts/console_script_create_sector_image.sh] 
test_full_test.py::test_integ_test_script[$geoips_repopath/tests/scripts/ami.tc.WV.geotiff.sh] 
[gw5] [  3%] FAILED test_full_test.py::test_integ_test_script[$geoips_repopath/tests/scripts/console_script_list_available_plugins.sh] 
test_full_test.py::test_integ_test_script[$geoips_repopath/tests/scripts/abi.static.Visible.imagery_annotated.sh] 
[gw4] [  5%] PASSED test_full_test.py::test_integ_test_script[$geoips_repopath/docs/build_docs.sh $data_fusion_repopath $data_fusion_pkgname html_only] 
```

...

``` bash

Fri Nov 8 22:52:37 UTC 2024  Running ahi.Cloud-Phase.imagery_clean.sh
/app/geoips_packages/geoips_clavrx/tests/scripts/ahi.Cloud-Phase.imagery_clean.sh
/output/logs/20241108/20241108.224757_geoips_clavrx/test_all_geoips_clavrx.log_ahi.Cloud-Phase.imagery_clean.sh.log
        Return: 0

Fri Nov 8 22:52:50 UTC 2024  Running ahi.Effective-Radius.imagery_clean.sh
/app/geoips_packages/geoips_clavrx/tests/scripts/ahi.Effective-Radius.imagery_clean.sh
/output/logs/20241108/20241108.224757_geoips_clavrx/test_all_geoips_clavrx.log_ahi.Effective-Radius.imagery_clean.sh.log
        Return: 0

Fri Nov 8 22:53:04 UTC 2024  Running ahi.Cloud-Temp-ACHA.imagery_clean.sh
/app/geoips_packages/geoips_clavrx/tests/scripts/ahi.Cloud-Temp-ACHA.imagery_clean.sh
/output/logs/20241108/20241108.224757_geoips_clavrx/test_all_geoips_clavrx.log_ahi.Cloud-Temp-ACHA.imagery_clean.sh.log
        Return: 0

Fri Nov 8 22:53:17 UTC 2024  Running ahi.Cloud-Type.imagery_clean.sh
/app/geoips_packages/geoips_clavrx/tests/scripts/ahi.Cloud-Type.imagery_clean.sh
/output/logs/20241108/20241108.224757_geoips_clavrx/test_all_geoips_clavrx.log_ahi.Cloud-Type.imagery_clean.sh.log
        Return: 0

Fri Nov 8 22:53:30 UTC 2024  Running ahi.Temp-11p0.imagery_clean.sh
/app/geoips_packages/geoips_clavrx/tests/scripts/ahi.Temp-11p0.imagery_clean.sh
/output/logs/20241108/20241108.224757_geoips_clavrx/test_all_geoips_clavrx.log_ahi.Temp-11p0.imagery_clean.sh.log
        Return: 0

Fri Nov 8 22:53:42 UTC 2024  Running ahi.Temp-3p75.imagery_clean.sh
/app/geoips_packages/geoips_clavrx/tests/scripts/ahi.Temp-3p75.imagery_clean.sh
/output/logs/20241108/20241108.224757_geoips_clavrx/test_all_geoips_clavrx.log_ahi.Temp-3p75.imagery_clean.sh.log
        Return: 0


Fri Nov 8 22:53:55 UTC 2024  Running post, final results in /output/logs/20241108/20241108.224757_geoips_clavrx/test_all_geoips_clavrx.log

Package: geoips_clavrx
Total run time: 358 seconds
Number data types run: 14
Number data types failed: 1
Fri Nov  8 22:53:55 UTC 2024

============================================================== short test summary info ==============================================================
FAILED test_full_test.py::test_integ_test_script[$geoips_repopath/tests/utils/check_code.sh all $recenter_tc_repopath] - subprocess.CalledProcessError: Command '['bash', '/app/geoips_packages/geoips/tests/integration_tests/full_install.sh', 'exit_on_missing']' retu...
FAILED test_full_test.py::test_integ_test_script[$geoips_repopath/docs/build_docs.sh $template_basic_plugin_repopath $template_basic_plugin_pkgname html_only] - subprocess.CalledProcessError: Command '['bash', '/app/geoips_packages/geoips/tests/integration_tests/full_install.sh', 'exit_on_missing']' retu...
FAILED test_full_test.py::test_integ_test_script[$geoips_repopath/tests/scripts/console_script_create_sector_image.sh] - subprocess.CalledProcessError: Command '['bash', '/app/geoips_packages/geoips/tests/integration_tests/full_install.sh', 'exit_on_missing']' retu...
FAILED test_full_test.py::test_integ_test_script[$geoips_repopath/tests/scripts/ami.tc.WV.geotiff.sh] - subprocess.CalledProcessError: Command '['bash', '/app/geoips_packages/geoips/tests/integration_tests/../../tests/scripts/ami.tc.WV.geotiff.sh']...
FAILED test_full_test.py::test_integ_test_script[$geoips_repopath/tests/scripts/amsr2.config_based_overlay_output.sh] - subprocess.CalledProcessError: Command '['bash', '/app/geoips_packages/geoips/tests/integration_tests/../../tests/scripts/amsr2.config_based_ove...
FAILED test_full_test.py::test_integ_test_script[$geoips_clavrx_repopath/tests/test_all.sh] - subprocess.CalledProcessError: Command '['bash', '/app/geoips_packages/geoips_clavrx/tests/test_all.sh']' returned non-zero exit status 1.
===================================================== 6 failed, 47 passed in 497.16s (0:08:17) ======================================================

```

**Tests that are failing in examples can be ignored**

### CODE REVIEW

File: tests/integration_tests/test_full_test.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+import subprocess
+import pytest
+import sys
+
+integ_test_calls = [
+    "$geoips_repopath/tests/utils/check_code.sh all $recenter_tc_repopath",
+    "$geoips_repopath/docs/build_docs.sh $geoips_repopath $geoips_pkgname html_only",
+    "$geoips_repopath/docs/build_docs.sh $recenter_tc_repopath $recenter_tc_pkgname html_only",  # noqa
+    "$geoips_repopath/docs/build_docs.sh $data_fusion_repopath $data_fusion_pkgname html_only",  # noqa
+    "$geoips_repopath/docs/build_docs.sh $template_basic_plugin_repopath $template_basic_plugin_pkgname html_only",  # noqa
```
Comment (mindyls) [inline tests/integration_tests/test_full_test.py:16]: As discussed, let's merge this PR, then open another branch/PR to remove full_test.sh (only use this one), and do any final cleanup on this PR.  I don't think we want to maintain both full_test.sh and test_full_test.py.

**Review by mindyls — APPROVED** [review]
Open another Issue/branch/PR to remove full_test.sh!

File: tests/integration_tests/test_full_test.py (hunk 1 above)
Comment (biosafetylvl5) [reply tests/integration_tests/test_full_test.py:16]: Do you want to also get rid of base_test? It'd be easy to put both in the pytest file, and then put "marks" so they can be run separately

eg. `pytest -m integration full` or `pytest -m integration base`

Comment (biosafetylvl5) [reply tests/integration_tests/test_full_test.py:16]: Issue created! :)

**Review by jsolbrig — APPROVED** [review]

**Pull request merged**